src/wallet/wallet.cpp

```diff
@@ -2513,27 +2520,34 @@ TransactionError CWallet::FillPSBT(PartiallySignedTransaction& psbtx, bool& comp
             script = input.non_witness_utxo->vout[txin.prevout.n].scriptPubKey;
         } else {
             // There's no UTXO so we can just skip this now
-            complete = false;
             continue;
         }
         SignatureData sigdata;
         input.FillSignatureData(sigdata);
-        std::unique_ptr<SigningProvider> provider = GetSigningProvider(script, sigdata);
-        if (!provider) {
-            complete = false;
+        ScriptPubKeyMan* spk_man = GetScriptPubKeyMan(script, sigdata);
+        if (!spk_man) {
             continue;
         }
 
-        complete &= SignPSBTInput(HidingSigningProvider(provider.get(), !sign, !bip32derivs), psbtx, i, sighash_type);
-    }
+        // If we've already been signed by this spk_man, skip it
```

Comment:
Will one SPKM always be sufficient for any input even going forward? Seems to me the same SPKM will always be sourced for a particular input?

Reply:
Documented the assumption.